SoftLayer/managers/storage_utils.py

```diff
@@ -1003,6 +1003,93 @@ def prepare_duplicate_order_object(manager, origin_volume, iops, tier,
+    """
+
+    # Verify that the origin volume has not been cancelled
+    if 'billingItem' not in origin_volume:
+        raise exceptions.SoftLayerError(
+            "The origin volume has been cancelled; "
+            "unable to order modify volume")
+
+    # Ensure the origin volume is STaaS v2 or higher
+    # and supports Encryption at Rest
+    if not _staas_version_is_v2_or_above(origin_volume):
+        raise exceptions.SoftLayerError(
+            "This volume cannot be modified since it "
+            "does not support Encryption at Rest.")
+
+    # Validate the requested new size, and set the size if none was given
+    new_size = _validate_new_size(
+        origin_volume, new_size, volume_type)
+
+    # Get the appropriate package for the order
+    # ('storage_as_a_service' is currently used for modifying volumes)
+    package = get_package(manager, 'storage_as_a_service')
+
+    # Determine the new IOPS or new tier level for the volume, along with
+    # the type and prices for the order
+    origin_storage_type = origin_volume['storageType']['keyName']
+    if origin_storage_type == 'PERFORMANCE_BLOCK_STORAGE'\
+            or origin_storage_type == 'PERFORMANCE_BLOCK_STORAGE_REPLICANT'\
+            or origin_storage_type == 'PERFORMANCE_FILE_STORAGE'\
+            or origin_storage_type == 'PERFORMANCE_FILE_STORAGE_REPLICANT':
+        volume_is_performance = True
+        new_iops = _validate_new_performance_iops(
+            origin_volume, new_iops, new_size)
+        # Set up the price array for the order
+        prices = [
+            find_price_by_category(package, 'storage_as_a_service'),
+            find_saas_perform_space_price(package, new_size),
+            find_saas_perform_iops_price(package, new_size, new_iops),
+        ]
+
+    elif origin_storage_type == 'ENDURANCE_BLOCK_STORAGE'\
+            or origin_storage_type == 'ENDURANCE_BLOCK_STORAGE_REPLICANT'\
+            or origin_storage_type == 'ENDURANCE_FILE_STORAGE'\
+            or origin_storage_type == 'ENDURANCE_FILE_STORAGE_REPLICANT':
+        volume_is_performance = False
+        new_tier = _validate_new_endurance_tier(origin_volume, new_tier)
+        # Set up the price array for the order
+        prices = [
+            find_price_by_category(package, 'storage_as_a_service'),
+            find_saas_endurance_space_price(package, new_size, new_tier),
+            find_saas_endurance_tier_price(package, new_tier),
+        ]
+
+    else:
+        raise exceptions.SoftLayerError(
+            "Origin volume does not have a valid storage type "
+            "(with an appropriate keyName to indicate the "
+            "volume is a PERFORMANCE or an ENDURANCE volume)")
+
+    modify_order = {
+        'complexType': 'SoftLayer_Container_Product_Order_'
+                       'Network_Storage_AsAService_Upgrade',
+        'packageId': package['id'],
+        'prices': prices,
+        'volume': origin_volume,
+        #'useHourlyPricing' : 'true',
```

Comment:
I don't think this line is needed anymore, since the hourly vs. monthly billing will be handled automatically by the API now.